### PR TITLE
docs - filters

### PR DESCRIPTION
**Pull request by nllho**
Super gnarly topic... so many dependencies!

Main changes:
- Filters troubleshooting: consolidated info with docs.
- Data reference: moved out of SQL section and brought up to date (it's a great resource for data types, which is the root cause of many issues, filters or otherwise).
- SQL: updated info on field filters and troubleshooting.

These changes break some links in Learn. I've got them noted down -- will follow up when the 45 docs go out.

Gonna take one more pass just to look at formatting tomorrow, but otherwise this is ready for review.

### CODE REVIEW

File: docs/embedding/signed-embedding-parameters.md [hunk 1]
```diff
@@ -71,14 +71,11 @@ You can use locked parameters to display filtered data based on attributes captu
 
 Locked parameters will apply the selected filter values to your original dashboard or SQL question, but they won't be displayed as filter widgets on your embed. Locked parameters may also limit the values that are shown in your [embedded filter widgets](#adding-a-filter-widget-to-a-signed-embed).
 
-## Adding multiple values to a locked parameter
+## Locked parameters on dashboards with SQL questions
```
Comment (nllho) [inline docs/embedding/signed-embedding-parameters.md:74]: Got this wrong when I first wrote it.

File: docs/data-modeling/metadata-editing.md [hunk 2]
```diff
@@ -141,6 +141,15 @@ You can manually change the user interface for the filter to:
 
 ![Filter options](./images/filter-options.png)
 
+## Changing a search box filter to a dropdown filter
```
Comment (nllho) [inline docs/data-modeling/metadata-editing.md:144]: I couldn't decide where to put this info --- it used to live in the Filters troubleshooting guide.

It's about _setting up_ a filter (so there's a link on `/dashboards/filters.md` above), but the actual changes need to be done on the Data Model page.

Does this make sense, or is there a more intuitive way to organize this?

File: docs/embedding/signed-embedding-parameters.md [hunk 3]
```diff
@@ -43,13 +43,13 @@ For example, if your embedded dashboard has a filter called "Breakfast", and you
...
 
-To specify default values for more than one filter, separate them with ampersands (&):
+To specify default values for more than one filter, separate the filters with ampersands (&):
 
 ```
 your_embedding_url?breakfast=Scrambled_eggs&lunch=Grilled_cheese
 ```
 
-If the original dashboard's filter widget accepts multiple values (i.e., it's a [dropdown filter](../dashboards/filters.md#choosing-between-a-dropdown-or-autocomplete-for-your-filter), not a text box filter), you can set multiple default values:
+You can set multiple default values for a filter by separating the _values_ with ampersands (&):
```
Comment (jeff-bruemmer) [inline docs/embedding/signed-embedding-parameters.md:52]:
```suggestion
You can set multiple default values for a filter by separating the `key=value` pairs with ampersands (&):
```

File: docs/exploration-and-organization/data-model-reference.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+Click on the table you would like to explore further to see a description of it and a list of all the columns it has. Each table or column will only contain a description if your admin wrote something in the Data Model section of the Admin Panel.
+
+If you click on a column you’re interested in, you’ll see a description of its contents, as well as a list of sample values for columns that don't have a huge number of distinct values.
+
+![Column detail](./images/data-reference-column-detail.png)
+
+## Foreign keys
+
+You can find a list of table relationships ([foreign keys](https://www.metabase.com/glossary/foreign_key)) under **Connected to these tables**. Metabase will display this section in the sidebar if a Metabase admin has documented the relationship between your current table and another table in the database. If a foreign key relationship exists in your database, but doesn't show up in your Metabase, your Metabase admin will need to update your [Metabase data model](../data-modeling/metadata-editing.md).
```
Comment (jeff-bruemmer) [inline docs/exploration-and-organization/data-model-reference.md:35]:
```suggestion
You can find a list of table relationships ([foreign keys](https://www.metabase.com/glossary/foreign_key)) under **Connected to these tables**. If a Metabase admin has documented the relationship between your current table and another table in the database, Metabase will display this section in the sidebar. If a foreign key relationship exists in your database, but doesn't show up in your Metabase, your Metabase admin will need to update your [Metabase data model](../data-modeling/metadata-editing.md).
```

File: docs/troubleshooting-guide/filters.md [hunk 5]
```diff
@@ -1,72 +1,90 @@
...
-If you're using a [SQL variable][sql-variable-gloss] with the variable type "field filter", go to [Troubleshooting SQL variables][troubleshoot-sql-variables].
+## Dashboard filters
 
-## Is the dashboard filter actually connected to your question?
+1. Go to your dashboard.
+2. Click the **pencil** icon to go into edit mode.
+3. Click the **gear** icon beside your filter widget.
+4. Make sure you've selected a column for your filter under **Column to filter on**.
+5. If you can't find the right **Column to filter on**, or you're getting "No Results" when you apply the filter:
+   - Go to the _original question_.
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/filters.md:21]: > Go to the _original question_.

How?

File: docs/troubleshooting-guide/filters.md [hunk 6]
```diff
@@ -1,72 +1,90 @@
...
+   - Follow the troubleshooting steps under [Question filters](#question-filters).
 
-**Root cause:** The filter isn't connected to any cards on the dashboard, or connected to the wrong field.
+## Question filters
 
-**Steps to take:**
+1. Go to your question.
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/filters.md:26]: > 1. Go to your question.

Do we need this step? If so, we should be more specific on how to get from dashboard to question.

File: docs/troubleshooting-guide/filters.md [hunk 7]
```diff
@@ -1,72 +1,90 @@
...
+- [SQL field filter](../questions/native-editor/sql-parameters.md#the-field-filter-variable-type): make sure you're using the correct [field filter syntax](../questions/native-editor/sql-parameters.md#field-filter-syntax), then see [Troubleshooting SQL variables](./sql.md#field-filter-variables).
 
-**Steps to take**:
+**Explanation**
 
-1. Check that your SQL query contains at least [one variable][sql-variable-gloss] for the filter to insert the value. These can be plain variables, or [Field Filters][field-filter], with names enclosed in double curly braces `{% raw %}{{variable_name}}{% endraw %}`, typically in a `WHERE` clause.
-2. If these steps don’t fix your error, go to [Troubleshooting SQL variables][troubleshoot-sql-variables].
+When we first set up a filter, we need to link the filter to a column. Our filter won't work at all if we make the wrong assumptions about column's values or data type. Or, the filter might suddenly stop working if the column gets changed.
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/filters.md:46]:
```suggestion
When we first set up a filter, we need to link the filter to a column. If we make the wrong assumptions about a column's values or data type, the filter won't work at all. If a column changes, the filter might suddenly stop working.
```

File: docs/troubleshooting-guide/filters.md [hunk 8]
```diff
@@ -1,72 +1,90 @@
...
-3. Click the gear-icon to view all the field's settings.
-4. Set **Field Type** to "Category" and **Filtering on this field** to "A list of all values."
-5. Click the button **Re-scan this field** in the bottom.
+**Explanation**
 
-If you created the question in SQL, then you only get a dropdown if the filter is a Field Filter _and_ the Filtering on this field option is set to your preferred input type: A list of all values (dropdown list) _and_ the number of unique values is less than 100.
+Metabase needs to know the data type of a column in order to present you with a curated selection of filter types. Sometimes these columns are mistyped---if a column stores your numbers as strings, Metabase will only show you text or category filters (with options like "is", "is not") instead of number filters (with options like "greater than", "less than").
 
-## Has someone renamed or deleted columns in the database?
+Timestamps, in particular, are the root of all evil, so please be patient with your Metabase admin (or yourself!) when trying to get the data type right.
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/filters.md:74]: > Timestamps, in particular, are the root of all evil

We should probably have spacetimestamps.

**Review by jeff-bruemmer — APPROVED** [review]

**Pull request merged**